pulpcore/openapi/__init__.py

```
@@ -454,4 +456,11 @@ def get_schema(self, request=None, public=False):
         server_url = "http://localhost:24817" if not request else request.build_absolute_uri("/")
         result["servers"] = [{"url": server_url}]
 
-        return normalize_result_object(result)
+        schema = normalize_result_object(result)
+
+        if request and "bindings" not in request.query_params:
+            # Validation fails when path does not start with /
+            # But bindings require some paths without /
+            validate_schema(schema)
```

Is this function dependant on the `--validate` flag?

---

No, it isn't.
`--validate` calls this, but I put it here just to fail early

---

so it's for debugging only?

---

not exactly debugging (but yes, I'm testing what is the best way to be sure the schema is valid)
`--validate` will fail only if you do: `pulpcore-manager spectacular --validate`
`validate_schema` will give a `500` when you hit `/pulp/api/v3/docs/api.json`

---

I think there's a good reason for the explicit `--validate` flag on the commandline tool.